project/Build.scala

```diff
@@ -1419,4 +1424,85 @@ object Build extends sbt.Build {
           }
       )
   ).dependsOn(partest % "test", library)
+
+  lazy val scalaTestSuite: Project = Project(
+    id = "scalaTestSuite",
+    base = file("scala-test-suite"),
+    settings = commonSettings ++ myScalaJSSettings ++ Seq(
+      publishArtifact in Compile := false,
+
+      testOptions += Tests.Argument(TestFrameworks.JUnit, "-v", "-a"),
+
+      unmanagedSources in Test ++= {
+        def loadList(listName: String): Set[String] = {
+          val listsDir = (resourceDirectory in Test).value / scalaVersion.value
+          val buff = scala.io.Source.fromFile(listsDir / listName)
+          val lines = buff.getLines().collect {
+            case line if !line.startsWith("#") && line.nonEmpty => line
+          }
+          lines.toSet
+        }
+
+        val blackList: Set[String] = loadList("BlacklistedTests.txt")
+        val whiteList: Set[String] = loadList("WhitelistedTests.txt")
+
+        val jUnitTestsPath =
+          (artifactPath in fetchScalaSource in partest).value / "test" / "junit"
+
+        def listFiles(file: File, acc: List[File], listed: List[String],
+            listedErr: List[String]): (List[File], List[String], List[String]) = {
+          if (file.isDirectory) {
+            file.listFiles().foldLeft((acc, listed, listedErr)) { (foldAcc, f) =>
+              listFiles(f, foldAcc._1, foldAcc._2, foldAcc._3)
+            }
+          } else {
+            file.relativeTo(jUnitTestsPath) match {
+              case Some(relPath) if relPath.getName.endsWith(".scala") =>
+                val relativePathString = relPath.toString
+                val blacklisted = blackList(relativePathString)
+                if (!(blacklisted ^ whiteList(relativePathString)))
+                  (acc, listed, relativePathString :: listedErr)
+                else if (blacklisted)
+                  (acc, relativePathString :: listed, listedErr)
+                else
+                  (file :: acc, relativePathString :: listed, listedErr)
+
+              case _ =>
+                (acc, listed, listedErr)
+            }
+          }
+        }
+
+        val _ = (fetchScalaSource in partest).value
+        val (whitelistedFiles, listed, listedErr) =
```

Comment:
No, listed is the list of all Scala sources found in the directory. It will be more explicit once I use `(topLevelDir ** "*.scala").get`.